File: exercises/python_playwright/Scrollbars/scrollbars_playwright.py
from playwright.sync_api import Page


class ScrollbarsPage:
    def __init__(self, page: Page):
        self.page = page
        self.hiding_button = page.get_by_role("button", name="Hiding Button")

    def open(self) -> None:
        self.page.goto("http://uitestingplayground.com/scrollbars")

    def click_hiding_button(self) -> None:
        self.hiding_button.scroll_into_view_if_needed()
        self.hiding_button.click()

File: exercises/javascript_playwright/scrollbars/scrollbarsPlaywright.js
export class ScrollbarsPage {
    /**
     *
     * @param {import("@playwright/test").Page} page
     */
    constructor(page) {
        this.page = page;
        this.button = page.locator("#hidingButton");
    }

    async openExercisePage() {
        await this.page.goto("http://uitestingplayground.com/scrollbars");
    }

    async clickButton() {
        await this.button.scrollIntoViewIfNeeded();
        await this.button.click();
    }
}
